packages/headless/src/controllers/controller/headless-controller.ts

```diff
@@ -35,6 +35,7 @@ export function buildController<T extends object>(
   return {
     subscribe(listener: () => void) {
       listener();
+      prevState = JSON.stringify(this.state);
```

Comment:
I found a somehow important bug debugging, the prevState was set to `{}` initially. Usually no state is initially an empty object, but contains properties. That means that the listener was called:
1. instantly, to set the state correctly (line 37)
2.  the first time the store's state updated (since the subscribe) in order to set prevState to a meaningful initial value
3.  anytime the scoped controller state is updated

The 2 should not happen, this is solved by setting prevState to the actual initial state right when subscribe is called.